python/cudf/cudf/tests/test_joining.py

```diff
@@ -735,3 +735,56 @@ def test_join_empty_table_dtype():
     pd_merge = left.merge(right, how="left", left_on=["a"], right_on=["b"])
     gd_merge = gleft.merge(gright, how="left", left_on=["a"], right_on=["b"])
     assert_eq(pd_merge["a"].dtype, gd_merge["a"].dtype)
+
+
+@pytest.mark.parametrize("how", ["outer", "inner", "left", "right"])
+@pytest.mark.parametrize("cat", [True, False])
+def test_join_multi(how, cat):
+
+    size = 10
+    index = ["b", "c"]
+    df1 = pd.DataFrame(
+        {
+            "a1": np.arange(size),
+            "b": np.random.randint(0, 2, size=size),
+            "c": np.random.choice(["dog", "cat"], size=size),
+        }
+    )
+    if cat:
+        df1["c"] = df1["c"].astype("category")
+    df1 = df1.set_index(index)
+    gdf1 = cudf.from_pandas(df1)
```

Comment:
`cudf.from_pandas` forwards to `cudf.Dataframe.from_pandas`, which has `nans_to_nulls=True`. If you include null-value tests, that might need to be accounted for.